torchdata/dataloader2/reading_service.py

```diff
@@ -174,38 +188,72 @@ def __init__(
         self.multiprocessing_context = multiprocessing_context
         self.processes = []
         self.datapipes = []
-        self.combined_datapipes = None
+        self.end_datapipe = None
+        self._pg = None
+        self._world_size = 1
+        self._rank = 0
 
     @staticmethod
-    def init_datapipe_process(num_workers, worker_id, datapipe):
-        # TODO(614): Add distributed support
-        # TODO(615): Add shuffle determinism support
-        torch.utils.data.graph_settings.apply_sharding(datapipe, num_workers, worker_id)
+    def _process_init_fn(world_size, rank, num_workers, worker_id, datapipe):
+        global_worker_id = worker_id * world_size + rank
+        total_num_workers = num_workers * world_size
+        torch.utils.data.graph_settings.apply_sharding(datapipe, total_num_workers, global_worker_id)
 
     @staticmethod
-    def call_on_epoch_reset(datapipe, *args):
+    def _process_reset_fn(world_size, rank, num_workers, worker_id, datapipe, shared_seed, *args):
```

Comment:
Feel free to drop `*args` here, as you accept just datapipe and seed.

Reply:
For sure. And, those APIs and messages are private, so I think it's totally fine to remove.

Reply:
nit: `num_workers` also isn't needed here

Reply:
> nit: `num_workers` also isn't needed here

Yeah, I intentionally add it here to align with those two static functions for now. With `SequentialRS`, we only need `worker_id`, `datapipe` and a `seed_generator` here.